docs/core_apis.md

```diff
@@ -14,15 +14,15 @@ curl -X POST http://localhost:4900/shutdown
 
 The following APIs are available:
 
-* /list
+* GET /list
     * lists running pipelines
-* /shutdown
+* POST /shutdown
     * starts a graceful shutdown of the Data Prepper
-* /metrics/prometheus
+* GET /metrics/prometheus
```

Comment:
We may want to double check to be sure that Prometheus use HTTP GET here. It seems reasonable. But, they might also use `POST`. I'm not sure.

Reply:
We do support both GET and POST in this PR for /metrics/prometheus; I can add POST here, but a few things I'm wondering about:
- I thought this is our own endpoint, why do we care what HTTP method Prometheus uses?
- Currently for paths /list, /metrics/sys and /metrics/prometheus, handling a POST request is nothing different from handling a GET request; it doesn't use anything from the request body. What's the reason to support POST here?
- /metrics/prometheus and /metrics/sys get routed to the same handler and I didn't see any difference in response.

Reply:
I think we can resolve this comment. Your change does retain `POST`. I don't think you need to make any changes regarding this.

> I thought this is our own endpoint, why do we care what HTTP method Prometheus uses?

This is an endpoint that we provide which Prometheus can scrape from to get metrics.

Reply:
Updated the doc to include supported HTTP methods in [4449ffd](https://github.com/opensearch-project/data-prepper/pull/1697/commits/4449ffd266f6b8026b3c4edfcfd14cab2dbe5be3)